src/transformers/models/sam_hq/modeling_sam_hq.py

```diff
@@ -0,0 +1,1790 @@
+            batch_size, point_batch_size, num_channels, height * width
+        )
+
+        masks_sam = (hyper_in[:, :, : self.num_mask_tokens - 1] @ upscaled_embedding).reshape(
+            batch_size, point_batch_size, -1, height, width
+        )
+        masks_hq = (hyper_in[:, :, self.num_mask_tokens - 1 :] @ upscaled_embedding_hq).reshape(
+            batch_size, point_batch_size, -1, height, width
+        )
+        masks = torch.cat([masks_sam, masks_hq], dim=2)
+
+        iou_pred = self.iou_prediction_head(iou_token_out)
+
+        if multimask_output:
+            mask_slice = slice(1, self.num_mask_tokens - 1)
+            iou_pred = iou_pred[:, :, mask_slice]
+            # Sort the IoU scores in descending order and get indices
+            iou_pred_sorted, sort_indices = torch.sort(iou_pred, dim=2, descending=True)
+            # Reorder the masks according to sorted scores
+            masks_sam = masks[:, :, mask_slice, :, :]
+            masks_sam = torch.gather(
+                masks_sam,
+                2,
+                sort_indices[..., None, None].expand(-1, -1, -1, masks_sam.shape[3], masks_sam.shape[4]),
+            )
+            # Update iou_pred with sorted scores
+            iou_pred = iou_pred_sorted
+        else:
+            mask_slice = slice(0, 1)
+            iou_pred = iou_pred[:, :, mask_slice]
+            masks_sam = masks[:, :, mask_slice, :, :]
+
+        masks_hq = masks[:, :, slice(self.num_mask_tokens - 1, self.num_mask_tokens), :, :]
+        if hq_token_only:
+            masks = masks_hq
+        else:
+            masks = masks_sam + masks_hq
+
+        outputs = (masks, iou_pred)
+        if output_attentions:
+            outputs = outputs + (attentions,)
+        else:
+            outputs = outputs + (None,)
+
+        return outputs
+
+
+class SamHQPreTrainedModel(PreTrainedModel):
+    config_class = SamHQConfig
+    base_model_prefix = "sam_hq"
+    main_input_name = "pixel_values"
+    _no_split_modules = ["SamHQVisionAttention"]
+    supports_gradient_checkpointing = True
+    _supports_sdpa = True
+
+    def _init_weights(self, module):
+        std = self.config.initializer_range
+        if isinstance(module, (nn.Linear, nn.Conv2d, nn.ConvTranspose2d)):
+            module.weight.data.normal_(mean=0.0, std=std)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.Embedding):
+            module.weight.data.normal_(mean=0.0, std=std)
+            if module.padding_idx is not None:
+                module.weight.data[module.padding_idx].zero_()
+        elif isinstance(module, (SamHQLayerNorm, nn.LayerNorm)):
+            module.weight.data.fill_(1.0)
+            module.bias.data.zero_()
+        elif isinstance(module, SamHQVisionAttention):
+            if module.use_rel_pos:
+                module.rel_pos_h.data.zero_()
+                module.rel_pos_w.data.zero_()
```

Comment:
Tests are broken by https://github.com/huggingface/transformers/pull/37070 which ensure weight initialization is done properly :sweat_smile: I think the following ought to fix it, can you try?
```suggestion
        elif isinstance(module, SamHQVisionAttention):
            if module.use_rel_pos:
                module.rel_pos_h.data.zero_()
                module.rel_pos_w.data.zero_()
        elif isinstance(module, SamHQVisionEncoder):
            if module.pos_embed is not None:
                module.pos_embed.data.zero_()
```

Reply:
Lol,I was beating my head around ,where I want wrong while merging with main branch

Reply:
add this in modular file